app/controllers/accounts/connected_accounts_controller.rb

```diff
@@ -16,7 +18,15 @@ def show
         sp_name: decorated_sp_session.sp_name,
         user: current_user,
         locked_for_session: pii_locked_for_session?(current_user),
+        change_email_available: change_email_available?,
       )
     end
+
+    def change_email_available?
+      if decorated_sp_session.requested_attributes.present?
```

Comment:
This seems to be looking at the requested attribute for _the current request_, not for each connected application. We want the link to be available for any connection where the requested attribute includes email.